savu/plugins/reconstructions/fista_recon.py

```diff
@@ -0,0 +1,170 @@
+    
+    def pre_process(self):
+        # extract given parameters
+        self.iterationsFISTA = self.parameters['iterationsFISTA']
+        self.datafidelity = self.parameters['datafidelity']
+        self.converg_const = self.parameters['converg_const']
+        self.regularisation = self.parameters['regularisation']
+        self.regularisation_parameter = self.parameters['regularisation_parameter']
+        self.regularisation_parameter2 = self.parameters['regularisation_parameter2']
+        self.regularisation_iterations = self.parameters['regularisation_iterations']
+        self.time_marching_parameter = self.parameters['time_marching_parameter']
+        self.edge_param = self.parameters['edge_param']
+        self.NDF_penalty = self.parameters['NDF_penalty']
+        self.Rectools = None
+
+        # extract data and data-related parameters
+        # in_data = self.get_in_datasets()[0]
+        # print(in_data.get_shape())
+        
+        #in_pData = self.get_plugin_in_datasets()[0]
+        
+        # get the shape of the input data
+        #anglesTot, DetectorsDimH = in_pData.get_shape()
+        
+        # need to extract angles to initiate reconstruction class
+        # mData = self.get_in_meta_data()[0]
+        # see all elements of the metadata
+        #for attr in dir(mData):
+        #    print("obj.%s = %r" % (attr, getattr(mData, attr)))
+        # angles = mData.get('rotation_angle').astype(np.float32)
+        # self.angles = np.deg2rad(angles) # get angles in radians
+
+        #print(self.Lipschitz_const)
+    def process_frames(self, data):
+        centre_of_rotations, angles, self.vol_shape, init  = self.get_frame_params()
+        sino = data[0].astype(np.float32)
+        anglesTot, self.DetectorsDimH = np.shape(sino)
+        self.angles = np.deg2rad(angles.astype(np.float32))
+        
+        # check if the reconstruction class has been initialised and calculate Lipschitz const
+        self.get_value() 
+        # recon = np.zeros([self.vol_shape,1,self.vol_shape])
+        
+        # Run FISTA reconstrucion algorithm
+        recon = self.Rectools.FISTA(sino,\
+                                    iterationsFISTA = self.iterationsFISTA,\
+                                    regularisation = self.regularisation,\
+                                    regularisation_parameter = self.regularisation_parameter,\
+                                    regularisation_iterations = self.regularisation_iterations,\
+                                    regularisation_parameter2 = self.regularisation_parameter2,\
+                                    time_marching_parameter = self.time_marching_parameter,\
+                                    edge_param = self.edge_param,\
+                                    NDF_penalty = self.NDF_penalty,\
+                                    lipschitz_const = self.Lipschitz_const)
+        return recon
+
+    def get_value(self):
```

Comment:
Given this use, i would probably call this something like setup_Lipschitz_constant